modules/network_synchronizer/data_buffer.cpp

```diff
@@ -277,7 +278,7 @@ real_t DataBuffer::add_real(real_t p_input, CompressionLevel p_compression_level
 	ERR_FAIL_COND_V(is_reading == true, p_input);
 
 	const real_t integral = Math::floor(p_input);
-	const real_t fractional = p_input - integral;
+	const real_t fractional = Math::fmod(p_input, 1);
 
 	const real_t added_integral = add_int(integral, p_compression_level);
 	const real_t added_fractional = add_unit_real(fractional, COMPRESSION_LEVEL_1);
```

Comment:
No need to store the sign here, we can just use `add_positive_unit_real`. The integral part already has this info.